src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,78 @@
+package com.endava.internship.mocking.repository;
+
+import com.endava.internship.mocking.model.Payment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+class InMemPaymentRepositoryTest {
+
+    private InMemPaymentRepository paymentRepository;
+    private Payment payment;
+    private UUID uuid;
+    private Optional<Payment> optional;
+
+    @BeforeEach
+    void setup(){
```

Comment:
We can define and init the objects in the class as fields;
If you want to have something dynamic, you can leave here only that particular code.